src/app/pages/homepage/homepage.component.ts

```diff
@@ -25,24 +26,33 @@ export class HomepageComponent implements OnInit {
 
   constructor(
     public authservice: AuthService,
-    public router: Router
+    public messageService: MessageService
   ) { }
 
   public ngOnInit() {
+    this.getCurrentEmail();
+    this.messageService.loadChat();
+    setInterval(function() {
+      this.messageService.loadChat();
+    }, 300000);
+    this.messageService.newsBlockfunc();
   }
-
-  public sendMessage(text: string) {
-    const author1: any = { name: 'me', id: 1 };
-    const author2: any = { name: 'Some idiot', id: 2 };
-    const id: number = this.chatMessages.length + 1;
-    const message: any = {
-      text,
-      id,
-      author: !(id % 2) ? author1.name : author2.name,
-      timestamp: new Date().toISOString()
-    };
-    this.chatMessages.push(message);
-    this.chat.message = '';
+  public openInput() {
+    if ( this.newsInputOpen ) {
+      this.newsInputOpen = false;
+    } else { this.newsInputOpen = true; }
+  }
+  public getCurrentUser() {
+    // Retrieve data by key from local storage
+    const userString: string = localStorage.getItem('currentUser');
+    // Return user object if data in local storage exist, or empty object if no user data available
+    return typeof userString === 'string' ? JSON.parse(userString) : {};
+  }
+  public getCurrentEmail() {
+    const userEmail: any = this.getCurrentUser();
+    this.userEmail = userEmail.email;
+    this.adminUserId = (userEmail.id === 1) ? true : false;
```

Comment:
Instead of previous 2 lines you shoud apply user object to some "public" variable like "appUser" and only check if this user role is "admin" or "super", but not to make "magic adminUserId boolean public property")))))